### PR TITLE
Improving lang.choice test interval

### PR DESCRIPTION
**Pull request by vluzrmos**
#15 #25 

Correct branch!

### CODE REVIEW

File: src/js/lang.js [hunk 1]
```diff
@@ -249,8 +249,91 @@
...
          */
+        
+        var numbers = this._parseNumbersFromInterval(interval);
+        
+        var types = {
+            "setOfNumbers" : /^\{.*\}$/,
+            "bothExclusive": /^(\(|\]|\)).*(\)|\[|\()$/,
+            "bothInclusive": /^\[.*\]$/,
+            "leftInclusive": /^\[.*(\)|\[|\()$/,
+            "rightInclusive": /^(\(|\]|\)).*\]$/
```
Comment (rmariuzzo) [inline src/js/lang.js]: @vluzrmos instead of `"` use `'` for strings.

File: src/js/lang.js [hunk 2]
```diff
@@ -249,8 +249,91 @@
...
+        var total = numbers.length;
+
+        var braces = /\[|\]|\{|\}|\(|\)/g;
+
+        /* first number */
+        var first  = numbers[0].replace(braces, '');
+        
+        numbers[0] = first;
+        
+        var last = '';
```
Comment (rmariuzzo) [inline src/js/lang.js]: @vluzrmos can we put all variable declaration together without new lines separating them? Something like:

```js
var numbers = interval.split(/,\s?/);
var total = numbers.length;
var braces = /\[|\]|\{|\}|\(|\)/g;
var first  = numbers[0].replace(braces, '');
```

Also, I think `var last = '';` can be put together.

File: src/js/lang.js [hunk 3]
```diff
@@ -249,8 +249,91 @@
...
+
+        if (total > 1) {
+            last = numbers[total-1].replace(braces, '');
+
+            numbers[total-1] = last;
```
Comment (rmariuzzo) [inline src/js/lang.js]: @vluzrmos same here, can we remove this spare new line? =)

File: tests/spec/lang_choice_spec.js [hunk 4]
```diff
@@ -26,6 +26,16 @@ describe('The Lang.choice() method', function() {
...
+    it('should count correctly', function() {
+        expect(Lang.choice('messages.count', 0)).toBe('There are none');
+        expect(Lang.choice('messages.count', 1)).toBe('There are some');
+        expect(Lang.choice('messages.count', 10)).toBe('There are some');
+        expect(Lang.choice('messages.count', 19)).toBe('There are some');
+        expect(Lang.choice('messages.count', 19)).toBe('There are some');
+        expect(Lang.choice('messages.count', 20)).toBe('There are many');
+        expect(Lang.choice('messages.count', 30)).toBe('There are many');
+    });
+
```
Comment (rmariuzzo) [inline tests/spec/lang_choice_spec.js]: @vluzrmos good stuff!

**Comment by rmariuzzo** [comment]
@vluzrmos all look good to me. Tell me what you think of the suggestions and apply them, then I will merge it.

**Comment by vluzrmos** [comment]
Thanks, I will do this...

**Comment by vluzrmos** [comment]
@rmariuzzo done!

File: tests/spec/lang_choice_spec.js (hunk 4 above)
Comment (vluzrmos) [inline tests/spec/lang_choice_spec.js]: My first impressions at Jasmine, I think it is awesome! :D

**Comment by vluzrmos** [comment]
Now it's better. I think you should merge. :+1:
@rmariuzzo

**Comment by vluzrmos** [comment]
I released 1 new npm / bower package with that script, if you want I could merge it on Lang.js
https://github.com/vluzrmos/interval-js

I think that is more compreensive way and will reduce the LangJs responsability.

**Pull request merged**